buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -250,24 +253,24 @@ class PerformanceTestPlugin : Plugin<Project> {
         prepareSamplesTask: TaskProvider<Task>
     ) {
 
-        fun create(name: String, configure: DistributedPerformanceTest.() -> Unit = {}) {
-            createDistributedPerformanceTestTask(name, performanceSourceSet, prepareSamplesTask).configure(configure)
+        fun create(name: String, clazz: KClass<out DistributedPerformanceTest>, configure: DistributedPerformanceTest.() -> Unit = {}) {
+            createDistributedPerformanceTestTask(name, clazz, performanceSourceSet, prepareSamplesTask).configure(configure)
         }
 
-        create("distributedPerformanceTest") {
+        create("distributedPerformanceTest", RerunableDistributedPerformanceTest::class) {
```

Comment:
I wonder if it would be better to be able to signal that the performance test will be re-run from TC to the perfomance test instead of using a different task class here. That way, the logic which tests will be re-run and which not would only be in one place - the TC config.

WDYT?

Reply:
I tried adding a `hasRerun` to `DistributedPerformanceTest` but gave up finally. It adds too many logic in `DistributedPerformance` (conditionally generate report, conditionally generate result json, write binary result, etc.) - I think `DistributedPerformance` is already huge enough.

Reply:
It would really like to avoid having the logic which performance test tasks to re-run in two places. Can we somehow avoid that?

Would it be very bad to generate the report twice? I was under the impression that we should always generate the report on the first run, and only re-generate the report on the second one if there were failed tests.

Reply:
> It would really like to avoid having the logic which performance test tasks to re-run in two places.

I'd like it, too, but I don't want to add more logic to `DistributedPerformanceTest`.

> Would it be very bad to generate the report twice?

One thing I can think of is that it usually takes over 1 minute to generate a report:

![image](https://user-images.githubusercontent.com/12689835/55336597-c2857380-54cf-11e9-9e16-01a6c0d0e632.png)

I think this is a quite expensive operation.

Reply:
For the re-run, we don't need to regenerate all the scenario graphs, only the ones which have been re-executed. And the overview. So maybe this would be much faster than generating all 100 graphs.